data_sync/mvds.md

```diff
@@ -0,0 +1,120 @@
+# Minimum Viable Data Synchronization
+
+**DRAFT VERSION 1.0**
+
+*Written by Oskar Thorén oskar@status.im & Dean Eigenmann dean@status.im*
+
+## Table of Contents
+
+1. [Abstract](#abstract)
+2. [Definitions](#definitions)
+3. [Wire Protocol](#wire-protocol)
+    1. [Transport](#transport) 
+    2. [Payloads](#payloads)
+4. [Synchronization](#synchronization)
+    1. [State](#state)
+    2. [Flow](#flow)
+    3. [Retransmission](#retransmission)
+5. [Footnotes](#footnotes)
+
+## Abstract
+
+In this specification we describe a minimum viable protocol for data synchronization inspired by the Bramble Synchronization Protocol<sup>1</sup>. This protocol is designed to ensure reliable messaging between peers communicating across an unreliable peer-to-peer (P2P) network where they may be unreachable or unresponsive.
+
+We present a functional specification for future implementation<sup>2</sup> as well as reference simulation data which demonstrates its performance.
+
+## Definitions
+
+| Term | Description |
+|------|-------------|
+| **Peer** | The other nodes that a node is connected to. |
+
+## Wire Protocol
+
+### Secure Transport
+
+This specification does not define anything related to the transport of packets. It is assumed that this is abstracted in such a way that any secure transport protocol could be easily implemented. Likewise, properties such as confidentiality, integrity, authenticity and forward secrecy are assumped to be provided by a layer below.
+
+### Payloads
+
+Payloads are implemented using [protocol buffers](https://developers.google.com/protocol-buffers/).
+
+```protobuf
+message Payload {
+  Ack ack = 1;
+  Offer offer = 2;
+  Request request = 3;
+  repeated Message messages = 4;
+}
+
+message Ack {
+  repeated bytes id = 1;
+}
+
+message Message {
+  bytes group_id = 1;
+  int64 timestamp = 2;
+  bytes body = 3;
+}
+
+message Offer {
+  repeated bytes id = 1;
+}
+
+message Request {
+  repeated bytes id = 1;
+}
+```
+
+## Synchronization
+
+### State
```

Comment:
The data layout for this isn't as clear as it can be, Can we qualify state? I.e. if we mean state wrt peer, message we can make this more explicit. Likewise for operations that
- touch all messages
- touch all peers